test/integration/buildruns_to_taskruns_test.go

```diff
@@ -342,6 +342,43 @@ var _ = Describe("Integration tests BuildRuns and TaskRuns", func() {
 		})
 	})
 
+	Context("when a standalone buildrun is created and the buildrun is cancelled", func() {
+
+		var standAloneBuildRunSample []byte
+		var standaloneBuildRunObject *v1alpha1.BuildRun
+
+		BeforeEach(func() {
+			standAloneBuildRunSample = []byte(test.MinimalOneOffBuildRun)
+
+			standaloneBuildRunObject, err = tb.Catalog.LoadStandAloneBuildRunWithNameAndStrategy(BUILDRUN+tb.Namespace+"-standalone", cbsObject, standAloneBuildRunSample)
+			Expect(err).To(BeNil())
+		})
+
+		It("should reflect a TaskRunCancelled reason and no completionTime", func() {
+			Expect(tb.CreateBR(standaloneBuildRunObject)).ToNot(HaveOccurred())
+
+			br, err := tb.GetBRTillStartTime(standaloneBuildRunObject.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = tb.GetTaskRunFromBuildRun(br.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			br.Spec.State = v1alpha1.BuildRunRequestedStatePtr(v1alpha1.BuildRunStateCancel)
+			data := []byte(fmt.Sprintf(`{"spec":{"state": "%s"}}`, v1alpha1.BuildRunStateCancel))
+			br, err = tb.BuildClientSet.ShipwrightV1alpha1().
+				BuildRuns(tb.Namespace).
+				Patch(tb.Context, br.Name, types.MergePatchType, data, metav1.PatchOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			actualReason, err := tb.GetTRTillDesiredReason(br.Name, "TaskRunCancelled")
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to get desired reason; expected %s, got %s", "TaskRunCancelled", actualReason))
+
+			reason, err := tb.GetBRReason(br.Name)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(reason).To(Equal(v1alpha1.BuildRunStateCancel))
```

Comment:
We have a race condition here. The TaskRun will be canceled and we verify that above and then the buildrun controller will have to reconcile on the TaskRun update and update the BuildRun status. This should be fast but let's not rely on it to happen instantanous.

```suggestion
			actualReason, err = tb.GetBRTillDesiredReason(br.Name, v1alpha1.BuildRunStateCancel)
			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to get desired BuildRun reason; expected %s, got %s", v1alpha1.BuildRunStateCancel, actualReason))
```